### PR TITLE
Don't assume apihost is https for sdk and action urls

### LINKED ISSUE #2719: wsk sdk install assumes https
```bash
> wsk property set --apihost http://192.168.99.100
> wsk sdk install docker
error: The docker SDK installation failed: Unable to retrieve 'blackbox-0.1.0.tar.gz' SDK: Get https://192.168.99.100/blackbox-0.1.0.tar.gz: x509: cannot validate certificate for 192.168.99.100 because it doesn't contain any IP SANs

> wsk sdk install ios
error: The ios SDK installation failed: Unable to retrieve 'OpenWhiskIOSStarterApp.zip' SDK: Get https://192.168.99.100/OpenWhiskIOSStarterApp.zip: x509: cannot validate certificate for 192.168.99.100 because it doesn't contain any IP SANs
```

### LINKED ISSUE #2720: cli is oblivous to protocol set for apihost
```bash
> wsk property set --apihost http://192.168.99.100
> wsk action create t echo.js --web true
ok: created action t
> wsk action get t --url
ok: got action t
https://http://192.168.99.100/api/v1/web/guest/default/t
```

### PR DESCRIPTION
**Pull request by bbrowning**
Reuse the getURLBase utility method when computing the URL for sdk
downloads and action URLs.

This fixes #2720 and fixes #2719.

### CODE REVIEW

**Comment by rabbah** [comment]
Thank you!

**Comment by rabbah** [comment]
@bbrowning I found some more https prefixing which I noted in this comment:

https://github.com/apache/incubator-openwhisk/issues/2720#issuecomment-329360303

**Comment by dubeejw** [comment]
PG2 2092 🔴

File: tools/cli/go-whisk/whisk/action.go [hunk 1]
```diff
@@ -122,33 +122,39 @@ func (action Action) WebAction() (webExportValue bool) {
 Returns the URL of an action as a string. A valid API host, path and version must be passed. A package that contains the
 action must be passed as well. An empty string must be passed if the action is not packaged.
  */
-func (action Action) ActionURL(apiHost string, apiPath string, apiVersion string, pkg string) (actionURL string) {
-    webActionPath := "https://%s%s/%s/web/%s/%s/%s"
-    actionPath := "https://%s%s/%s/namespaces/%s/actions/%s"
+func (action Action) ActionURL(apiHost string, apiPath string, apiVersion string, pkg string) (string, error) {
+    baseURL, err := GetURLBase(apiHost, apiPath)
+    if err != nil {
+        Debug(DbgError, "getURLBase(%s, %s) failed: %s\n", apiHost, apiPath, err)
```
Comment (dubeejw) [inline tools/cli/go-whisk/whisk/action.go]: Wrong debug message here. `getURLBase` should be `GetURLBase`.

File: tools/cli/go-whisk-cli/commands/commands.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ const DefaultOpenWhiskApiPath string = "/api"
 var UserAgent string = "OpenWhisk-CLI"
 
 func setupClientConfig(cmd *cobra.Command, args []string) (error){
-    baseURL, err := getURLBase(Properties.APIHost, DefaultOpenWhiskApiPath)
+    baseURL, err := whisk.GetURLBase(Properties.APIHost, DefaultOpenWhiskApiPath)
```
Comment (dubeejw) [inline tools/cli/go-whisk-cli/commands/commands.go]: Wrong debug message on line 48. `getURLBase` should be `GetURLBase`.

**Comment by dubeejw** [comment]
PG2 2093 ⏳

**Comment by dubeejw** [comment]
PG2 2093 🔵

File: tools/cli/go-whisk/whisk/action.go (hunk 1 above)
Comment (bbrowning) [reply tools/cli/go-whisk/whisk/action.go]: I was unsure if this was meant to be the exact method name in the debug message or if it just happened to match the old name. I'll update these.

**Pull request merged**